api/rosetta/data_network_integration_test.go

```diff
@@ -0,0 +1,523 @@
+	assert.Equal(t, status.CurrentBlockID.Hash, lastBlock.ID().String())
+	assert.Equal(t, status.CurrentBlockTimestamp, convert.RosettaTime(lastBlock.Timestamp))
+
+	// verify oldest block
+	assert.Equal(t, status.OldestBlockID.Hash, oldestBlockID)
+
+	// this is actually omitted from JSON, due to height being zero, and JSON having the 'omitempty' tag
+	assert.Equal(t, status.OldestBlockID.Index, uint64(0))
+}
+
+func TestAPI_StatusHandlesErrors(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	tests := []struct {
+		name string
+
+		request rosetta.StatusRequest
+
+		checkError assert.ErrorAssertionFunc
+	}{
+		{
+			name: "missing blockchain",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: "",
+					Network:    dps.FlowTestnet.String(),
+				},
+			},
+
+			checkError: checkRosettaError(http.StatusBadRequest, configuration.ErrorInvalidFormat),
+		},
+		{
+			name: "invalid blockchain",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: invalidBlockchain,
+					Network:    dps.FlowTestnet.String(),
+				},
+			},
+
+			checkError: checkRosettaError(http.StatusUnprocessableEntity, configuration.ErrorInvalidNetwork),
+		},
+		{
+			name: "missing network",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: dps.FlowBlockchain,
+					Network:    "",
+				},
+			},
+
+			checkError: checkRosettaError(http.StatusBadRequest, configuration.ErrorInvalidFormat),
+		},
+		{
+			name: "invalid network",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: dps.FlowBlockchain,
+					Network:    invalidNetwork,
+				},
+			},
+
+			checkError: checkRosettaError(http.StatusUnprocessableEntity, configuration.ErrorInvalidNetwork),
+		},
+	}
+
+	for _, test := range tests {
+
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			_, ctx, err := setupRecorder(statusEndpoint, test.request)
+			require.NoError(t, err)
+
+			err = api.Status(ctx)
+			test.checkError(t, err)
+		})
+	}
+}
+
+func TestAPI_Networks(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	// network request is basically an empty payload at the moment,
+	// there is a 'metadata' object that we're ignoring;
+	// but we can have the scaffolding here in case something changes
+
+	var netReq rosetta.NetworksRequest
+
+	rec, ctx, err := setupRecorder(listEndpoint, netReq)
+	require.NoError(t, err)
+
+	// execute the request
+	err = api.Networks(ctx)
+	assert.NoError(t, err)
+
+	var network rosetta.NetworksResponse
+	require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &network))
+
+	if assert.Len(t, network.NetworkIDs, 1) {
+		assert.Equal(t, network.NetworkIDs[0].Blockchain, dps.FlowBlockchain)
+		assert.Equal(t, network.NetworkIDs[0].Network, dps.FlowTestnet.String())
+	}
+}
+
+func TestAPI_Options(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	const errorCount = 9
```

Comment:
I'd suggest renaming this to `wantErrorCount` or something like that. In a test, a variable called `errorCount` sounds to me like it will be used to count errors, and not to compare with the actual number of errors we got.